mealz-shared-analytics/src/commonMain/kotlin/mealz/ai/AbstractSharedAnalytics.kt

```diff
@@ -34,11 +35,11 @@ abstract class AbstractSharedAnalytics {
         // Not using a list is the lightest way to do it when compiled in JS
         while (pathWithoutURL.isNotEmpty()) {
             // Find the index of the next '/' to separate path segments
-            val nextSlashIndex = pathWithoutURL.indexOf('/').takeIf { it != -1 } ?: pathWithoutURL.length
+            val nextSlashIndex = pathWithoutURL.indexOf('/').takeIf { it!=-1 } ?: pathWithoutURL.length
             val part = pathWithoutURL.substring(0, nextSlashIndex)
-            // If the part is not in the valid set of path segments, throw an exception
-            if (!validParts.contains("|$part|")) {
-                throw IllegalArgumentException("Invalid path : $part")
+            // If the part is nor in the valid set of path segments or a number, throw an exception
```

Comment:
You had the right idea to use nor 
```suggestion
            // If the part is not in the valid set of path segments nor a number, throw an exception
```

Reply:
thanks